### ISSUE TITLE
expose consumer heartbeat

### ISSUE THREAD

**Issue by filiptepper** [karafka/karafka#354]
For long running controllers, if session times out due to the lack of heartbeat, manual triggering of heartbeat would prevent consumer from being removed from consumer group.

```
def consume
...
long_running_method_a
send_heartbeat
long_running_method_b
send_heartbeat
long_running_method_c
end
```

**Comment by mensfeld** [comment]
I mean I can alias those exactly as the `#mark_as_consumed` is aliased but anyhow I wouldn't want to put this logic in the consumer itself.

**Comment by filiptepper** [comment]
@mensfeld I believe it will!

**Comment by mensfeld** [comment]
It will require a bit of work on the ruby-kafka side, but will take care of that.

**Comment by mensfeld** [comment]
Waiting for: https://github.com/zendesk/ruby-kafka/pull/593

**Comment by mensfeld** [comment]
@filiptepper small change in the API. Since the heartbeat is not public API, I don't want to "reach out" to it and will alias to the kafka client keeping ruby-kafka api (that I suggested and implemented btw ;) ):

```ruby
trigger_heartbeat
trigger_heartbeat!
```

this will be targeted to 1.3, for 1.2 it's a blocking only: #366

**Comment by mensfeld** [comment]
I will close that issue once proper heartbeating is merged to 1.3.
